scriptmodules/ports/splitwolf.sh

```diff
@@ -0,0 +1,124 @@
+    fi
+}
+
+function _get_opts_splitwolf() {
+    echo 'splitwolf-wolf3d VERSION_WOLF3D_SHAREWARE=y' # shareware v1.4
+    echo 'splitwolf-wolf3d_apogee VERSION_WOLF3D_APOGEE=y' # 3d realms / apogee v1.4 full
+    echo 'splitwolf-wolf3d_full VERSION_WOLF3D=y' # gt / id / activision v1.4 full
+    echo 'splitwolf-sod VERSION_SPEAR=y' # spear of destiny
+    echo 'splitwolf-sodmp VERSION_SPEAR_MP=y' # spear of destiny mission packs
+    echo 'splitwolf-spear_demo VERSION_SPEAR_DEMO=y' # spear of destiny
+}
+
+function build_splitwolf() {
+    mkdir "bin"
+    local opt
+    while read -r opt; do
+        local bin="${opt%% *}"
+        local defs="${opt#* }"
+        make clean
+        make $defs DATADIR="$romdir/ports/wolf3d/"
+        mv $bin "bin/$bin"
+        md_ret_require+=("bin/$bin")
+    done < <(_get_opts_splitwolf)
+}
+
+function install_splitwolf() {
+    cp gamecontrollerdb.txt bin/
+    cp -r lwmp bin/
+    md_ret_files=('bin')
+}
+
+function configure_splitwolf() {
+    local game
+
+    mkRomDir "ports/wolf3d"
+
+    # remove obsolete emulator entries
+    while read game; do
+        delEmulator "${game%% *}" "splitwolf"
+    done < <(_get_opts_splitwolf)
+
+    if [[ "$md_mode" == "install" ]]; then
+        game_data_wolf4sdl
+        cat > "$md_inst/bin/splitwolf.sh" << _EOF_
+#!/bin/bash
+
+function get_md5sum() {
+    local file="\$1"
+
+    [[ -n "\$file" ]] && md5sum "\$file" 2>/dev/null | cut -d" " -f1
+}
+
+function launch_splitwolf() {
+    local wad_file="\$1"
+    declare -A game_checksums=(
+        ['6efa079414b817c97db779cecfb081c9']="splitwolf-wolf3d"
+        ['a6d901dfb455dfac96db5e4705837cdb']="splitwolf-wolf3d_apogee"
+        ['b8ff4997461bafa5ef2a94c11f9de001']="splitwolf-wolf3d_full"
+        ['b1dac0a8786c7cdbb09331a4eba00652']="splitwolf-sod"
+        ['25d92ac0ba012a1e9335c747eb4ab177']="splitwolf-sodmp --mission 2"
+        ['94aeef7980ef640c448087f92be16d83']="splitwolf-sodmp --mission 3"
+        ['35afda760bea840b547d686a930322dc']="splitwolf-spear_demo"
+    )
+        if [[ "\${game_checksums[\$(get_md5sum \$wad_file)]}" ]] 2>/dev/null; then
+            $md_inst/bin/\${game_checksums[\$(get_md5sum \$wad_file)]} --splitdatadir $md_inst/bin/lwmp/ --split 2 --splitlayout 2x1
+        else
+            echo "Error: \$wad_file (md5: \$(get_md5sum \$wad_file)) is not a supported version"
+        fi
+}
+
+launch_splitwolf "\$1"
+_EOF_
+        chmod +x "$md_inst/bin/splitwolf.sh"
+    fi
+
+    declare -A games=(
+        ['vswap.wl1']="Splitwolf - Wolf 3D Demo"
+        ['vswap.wl6']="Splitwolf - Wolf 3D"
+        ['vswap.sod']="Splitwolf - Spear of Destiny Ep 1"
+        ['vswap.sd2']="Splitwolf - Spear of Destiny Ep 2"
+        ['vswap.sd3']="Splitwolf - Spear of Destiny Ep 3"
+        ['vswap.sdm']="Splitwolf - Spear of Destiny Demo"
+    )
+
+    add_games_wolf4sdl "$md_inst/bin/splitwolf.sh %ROM%" "$(declare -p games)"
```

Comment:
@joolswills this is where I pass `games` in `splitwolf.sh` -- if I defined it as `local` above, `add_games_wolf4sdl` would not be able to access it.

Reply:
it would be able to access it

bash local scope variables are usable in child functions - we use that for the $md_ variables which are passed to the modules - they are local in the parent.

Reply:
No, that doesn't work. If i use `declare -A` at the top of `splitwolf.sh`, I cannot access it inside of `splitwolf.sh`'s functions functions nor can I access it inside of `wolf4sdl.sh`'s functions, even if I do `declare -g -A` (global).

Honestly, we are way outside of my knowledge when it comes to bash scripting. What you're describing does not seem possible, and I do not see another way to accomplish this. If you can tell me precisely the code change you want to see, I'll gladly do it, but I've been reading about this for an hour and I cannot come up with another way.